ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/ChainHeadTracker.java

```diff
@@ -69,33 +86,79 @@ public CompletableFuture<BlockHeader> getBestHeaderFromPeer(final EthPeer peer)
         .setMessage("Requesting chain head info from {}...")
         .addArgument(peer::getLoggableId)
         .log();
-    final CompletableFuture<AbstractPeerTask.PeerTaskResult<List<BlockHeader>>>
-        bestHeaderFromPeerCompletableFuture = getBestHeaderFromPeerCompletableFuture(peer);
-    final CompletableFuture<BlockHeader> future = new CompletableFuture<>();
-    bestHeaderFromPeerCompletableFuture.whenComplete(
-        (peerResult, error) -> {
-          if (peerResult != null && !peerResult.getResult().isEmpty()) {
-            final BlockHeader chainHeadHeader = peerResult.getResult().get(0);
-            peer.chainState().update(chainHeadHeader);
-            future.complete(chainHeadHeader);
-            LOG.atDebug()
-                .setMessage("Retrieved chain head info {} from {}...")
-                .addArgument(
-                    () -> chainHeadHeader.getNumber() + " (" + chainHeadHeader.getBlockHash() + ")")
-                .addArgument(peer::getLoggableId)
-                .log();
-          } else {
-            LOG.atDebug()
-                .setMessage("Failed to retrieve chain head info. Disconnecting {}... {}")
-                .addArgument(peer::getLoggableId)
-                .addArgument(error != null ? error : "Empty Response")
-                .log();
-            peer.disconnect(
-                DisconnectMessage.DisconnectReason.USELESS_PEER_FAILED_TO_RETRIEVE_CHAIN_HEAD);
-            future.complete(null);
-          }
-        });
-    return future;
+
+    if (synchronizerConfiguration.isPeerTaskSystemEnabled()) {
+      return ethContext
+          .getScheduler()
+          .scheduleServiceTask(
+              () -> {
+                GetHeadersFromPeerTask task =
+                    new GetHeadersFromPeerTask(
+                        Hash.wrap(peer.chainState().getBestBlock().getHash()),
```

Comment:
I'm not sure what you mean. It is using the GetHeadersFromPeerTask constructor which takes a block hash.

If you're referring to the fact that it also takes a block number, I also don't like this pattern, as it's ambiguous exactly what that block number is used for, but this matches the old code (for now). In general, the pattern is to ignore the block number if the block hash is provided. This is something we can probably clean up later in a separate ticket.

Reply:
Yeah I was referring to the fact it also takes a block number. Would be quite simple to just add another constructor that takes a block hash without the block number to avoid confusion with how the task is being used.